Tasks/AzureRmWebAppDeployment/MSDeployUtility.ts

```diff
@@ -0,0 +1,139 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+import Q = require('q');
+import tl = require('vsts-task-lib/task');
+
+var regedit = require('regedit');
+var azureRmUtil = require('./AzureRMUtil.js');
+
+export function fileExists(path) {
+  try  {
+    return tl.stats(path).isFile();
+  }
+  catch (e) {
+    if (e.code == 'ENOENT') {
+      return false;
+    }
+    tl.debug("Exception tl.stats (" + path + "): " + e);
+    throw e;
+  }
+}
+
+export function getMSDeployCmdArgs(packageFile: string, webAppNameForMSDeployCmd: string, publishingProfile,
+                             removeAdditionalFilesFlag: boolean, excludeFilesFromAppDataFlag: boolean, takeAppOfflineFlag: boolean,
+                             virtualApplication: string, setParametersFile: string, additionalArguments: string) : string {
+
+    var msDeployCmdArgs = ' -verb:sync';
+    msDeployCmdArgs += ' -source:package=\'' + packageFile + '\'';
+    msDeployCmdArgs += ' -dest:auto,ComputerName=https://' + publishingProfile.publishUrl + '/msdeploy.axd?site=' + webAppNameForMSDeployCmd + ',';
+    msDeployCmdArgs += 'userName=' + publishingProfile.userName + ',Password=' + publishingProfile.userPWD + ',AuthType=Basic';
+
+    if (setParametersFile) {
+        msDeployCmdArgs += ' -setParamFile=' + setParametersFile;
+    }
+    
+    if (virtualApplication) {
+        msDeployCmdArgs += ' -setParam:name=\'IIS Web Application Name\',value=\'' + webAppNameForMSDeployCmd + '/' + virtualApplication + '\'';
+    }
+    else {
+        msDeployCmdArgs += ' -setParam:name=\'IIS Web Application Name\',value=\'' + webAppNameForMSDeployCmd + '\'';
+    }
+
+    if (!removeAdditionalFilesFlag) {
+        msDeployCmdArgs += ' -enableRule:DoNotDeleteRule';
+    }
+
+    if (takeAppOfflineFlag) {
+        msDeployCmdArgs += ' -enableRule:AppOffline';
+    }
+
+    if (excludeFilesFromAppDataFlag) {
+        msDeployCmdArgs += ' -skip:Directory=App_Data';
+    }
+    
+    if (additionalArguments) {
+        msDeployCmdArgs += ' ' + additionalArguments;
+    }
+
+    var userAgent = tl.getVariable("AZURE_HTTP_USER_AGENT");
+    if (userAgent) {
+        msDeployCmdArgs += ' -userAgent:' + userAgent;
+    }
+    tl.debug(tl.loc('ConstructedmsDeploycomamndlinearguments'));
+    return msDeployCmdArgs;
+}
+
+export async function executeMSDeployCmd(msDeployCmdArgs: string, publishingProfile, webAppUri: string) {
+    try {
+        if(webAppUri) {
+            tl.setVariable(webAppUri, publishingProfile.destinationAppUrl);
+        }
+        var msDeployPath = await getMSDeployFullPath();
+            var statusCode = await tl.exec(msDeployPath, msDeployCmdArgs, <any> {failOnStdErr: true});
+            if ( statusCode === 0 ) {
+                tl.debug(tl.loc('WebappsuccessfullypublishedatUrl0',publishingProfile.destinationAppUrl));                
+                var deploymentResult = await azureRmUtil.updateDeploymentStatus(publishingProfile, true);
```

Comment:
Should updating deployment status in Kudu be responsibility of executeMsDeployCmd?